### ISSUE TITLE
Loosing context after response comes back from watson

### ISSUE THREAD

**Issue by dipesh-gandhi** [watson-developer-cloud/botkit-middleware#131]
When I'm in a child node in dialog flow, I'm loosing context after message passes .after intercept. 

Per below, I can see context arrives in watsonMiddleware.after but when user responds, the context is gone in the .before. Do I need to manage context somehow outside, I thought it would be supported by middleware? 

watsonMiddleware.after: 

  {
	"intents": [
		{
			"intent": "weather",
			"confidence": 1
		}
	],
	"entities": [],
	"input": {
		"text": "weather"
	},
	"output": {
		"text": [
			"Okay, I see you are in Austin, Texas. Is this correct?"
		],
		"nodes_visited": [
			"node_1_1520607928207"
		],
		"log_messages": []
	},
	"context": {
		"conversation_id": "83842dba-7a4d-4f48-8f93-e6bd7746242a",
		"system": {
			"dialog_stack": [
				{
					"dialog_node": "node_1_1520607928207"
				}
			],
			"dialog_turn_counter": 1,
			"dialog_request_counter": 1,
			"_node_output_map": {
				"node_1_1520607928207": [
					0
				]
			}
		}
	}
}

watsonMiddleware.before:
{ workspace_id: '193992aa-61d9-4a71-bb49-cffd749f124d',
     input: { text: 'yes' } } }

**Comment by Naktibalda** [comment]
What storage middleware do you use?

**Comment by dipesh-gandhi** [comment]
I'm using Postgres

**Comment by dipesh-gandhi** [comment]
Okay I got it working. I didn't realize context had to be persisted and therefore did not know I needed context column in Users table. Once I added, the context is there and everything is good! 

In terms of performance it seems like lots of table updates for every response from Watson. It would be better to leverage in-memory cache for this part instead of persisting in db. It may be good for continuing conversation but not good for performance.
